15-connect-app-and-store/README.md

```diff
@@ -0,0 +1,341 @@
+
+Similarly to `emails` now being hydrated via the component's props, our actions shoudld be as well. By passing our actions through `mapDispatchToProps` each "action" is wrapped in a call to `dispatch()`. Lets import `deleteEmail` and pass it through `mapDispatchToProps`. Doing this will add `deleteEmail()` as a prop to the `<Page />`, so lets add it to our `propTypes` as well.
+```js
+// Page.js
+
+// previous imports
+
+import {deleteEmail as deleteEmailAction} from '../actions';
+
+class Page extends PureComponent {
+
+  static propTypes = {
+    emails: PropTypes.arrayOf(EMAIL_PROP_TYPE),
+    deleteEmail: PropTypes.func,
+  };
+
+  // initial state
+
+  // class methods
+
+  // render
+}
+
+export default connect(
+  //_mapStateToProps
+  (state) => ({emails: state}),
+
+  //_mapDispatchToProps
+  (dispatch) => ({
+    deleteEmail: () => dispatch(deleteEmailAction)
+})
+```
+
+Additionally, we can further optimize this call by taking advantage of a feature of `mapDispatchToProps()`. If an object made entirely of *action creators* is passed directly to `_mapDispatchToProps`, it will implicitly wrap each one in a call to`dispatch()`. This means we can rewrite the above as:
+```js
+// Page.js
+
+// imports
+
+class Page extends PureComponent {
+
+  // class methods
+
+}
+
+export default connect(
+  //_mapStateToProps
+  (state) => ({emails: state}),
+
+  //_mapDispatchToProps
+  {
+    deleteEmail: deleteEmailAction
+  }
+);
+```
+Now we can update our `_handleItemDelete()` to utilize the redux action coming through props, rather than our previous version. Addtionally, since we are using our Redux action, and `emails` is from our `store` we should no longer manually optimistically update our email state upon the action being completed. The `store` will handle that all on its own, so instead we can just focus on the UI behavior of resetting the `selectedEmail` to `-1`. So our `_handleItemDelete` should now look something like:
+
+```js
+
+class Page extends PureComponent {
+
+  // props
+
+  // initial state
+
+  // lifecycle methods
+
+  _handleItemDelete(emailId) {
+    this.props.deleteEmail(emailId)
+    // Also reset `selectedEmailId` since we're deleting it
+    this.setState({selectedEmailId: -1});
+  }
+
+  // helpers
+
+  // render
+}
+```
+
+With this, our `deleteEmail()` action should properly go through `dispatch` to eventually update our store.
+
+Within `<Page />` there are still many references to our previous version of "action-reducers" and `this.state.emails`. However, once these have all been replaced with Redux actions, the app should work just as it did before but now built on the scalability of Redux.
+
+
+## Exercises
+
+1. Replace the remaining  in `Page.js` with the actions from `actions/index.js`
+  - *hint*: Some of the `_handlers` just update `this.state.emails` upon success. In which case you can pass the action *itself* directly to the child component and remove the handler.
```

Comment:
It may be worth having an example of this in the tasks. Not a big deal since it's more of an optimization